lib/fluent/test/driver/base.rb

```diff
@@ -87,7 +87,7 @@ def run(timeout: nil, start: true, shutdown: true, &block)
 
           sleep_with_watching_threads = ->(){
             if @instance.respond_to?(:_threads)
-              @instance._threads.values.each{|t| t.join(0) }
+              @instance._threads.each_value{|t| t.join(0) }
```

Comment:
Looks like [this error](https://github.com/fluent/fluentd/pull/4302#issuecomment-1746490368) did not happen before.

I'm checking if this change has any impact.

Reply:
Indeed, it has the impact. I am trying to understand how this relates.

Reply:
It seems that it is caused by the fact that the code

```
              @instance._threads.each_value{|t| t.join(0) }
```

is accessing `@_threads` instance varible from `PluginHelperThread` mixin, that is otherwise guarded by `@_threads_mutex`, but not in this case.

I will study a code a bit more, but in the mean time, I will remove this affected chunk from the PR.

Thank You for your help, @daipom

Reply:
Thanks!
Looks like this error no longer occurs.

I don't understand why this change causes this error and why this error is specific to macOS...
If you find out anything, please let us know. Thanks!
In the meantime, how about adding some comments to this line, such as

```
# Can not use each_value here. Somehow, it makes tests on macOS unstable.
# https://github.com/fluent/fluentd/pull/4302#issuecomment-1746490368
```

For now, I think the current modifications are fine.
I will merge this later.